### PR TITLE
feat: text parser

### PR DESCRIPTION
**Pull request by ice-endymion**
## Description
New TextParser and TextSpanBuilder

## Additional Notes
- reimplement TextParser API
- add `TextSpanBuilder` to build `TextSpan` for `Text.rich`
- add taps for `HashTags`, `Mentions`, `Link's` across the app
- add test for `TextParser`
- fixed bugs related to rich text 

## Type of Change
- [x] Bug fix
- [x] New feature
- [x] Breaking change
- [x] Refactoring
- [x] Documentation
- [x] Chore

### CODE REVIEW

**Review by ice-alcides — DISMISSED** [review]

**Review by ice-brontes — APPROVED** [review]
Good job with tests.

File: lib/app/components/text_span_builder/text_span_builder.dart [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+/// textSpanBuilder.dispose();
+/// ```
+class TextSpanBuilder {
+  /// Creates a new instance of `TextSpanBuilder`.
+  ///
+  /// - [defaultStyle]: The default `TextStyle` to apply to text segments that do not have a specific style.
+  /// - [matcherStyles]: A map of `TextMatcher` instances to their corresponding `TextStyle`s.
+  ///   This allows you to specify custom styles for different matchers.
+  ///
+  /// If [matcherStyles] is not provided, it defaults to an empty map.
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/components/text_span_builder/text_span_builder.dart]: Simplified

File: lib/app/features/feed/views/components/post/components/post_body/post_body.dart [hunk 2]
```diff
@@ -21,19 +26,49 @@ class PostBody extends ConsumerWidget {
...
+      filteredContent,
+      onTap: (match) {
+        if (match.matcher is HashtagMatcher) {
+          FeedAdvancedSearchRoute(query: match.text).go(context);
+        }
+        if (match.matcher is UrlMatcher) openUrlInAppBrowser(match.text);
+      },
+    );
+
+    useEffect(
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart]: add

File: lib/app/features/feed/views/components/post/components/post_body/post_body.dart [hunk 3]
```diff
@@ -21,19 +26,49 @@ class PostBody extends ConsumerWidget {
...
+        matcherStyles: TextSpanBuilder.defaultMatchersStyles(context),
+      ),
+    );
+
+    final filteredContent =
+        (postMedia.isEmpty) ? postEntity.data.content : _excludeMediaLinks(postMedia);
+
+    final postText = textSpanBuilder.build(
+      filteredContent,
+      onTap: (match) {
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart:40]: Add `TextSpanBuilder.defaultOnTap(context, match: match)`, I don't add it as a default param for func cuz we need to have null onTap at least for profile bio, also it's good to have flexible api.

File: lib/app/features/feed/views/components/post/components/post_body/post_body.dart [hunk 4]
```diff
@@ -21,19 +26,49 @@ class PostBody extends ConsumerWidget {
...
           postText,
-          style: context.theme.appTextThemes.body2.copyWith(
-            color: context.theme.appColors.sharkText,
-          ),
         ),
       ],
     );
   }
+
+  List<TextMatch> _excludeMediaLinks(List<MediaAttachment> postMedia) {
```
[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart:59]: Not sure, we call `usePostMedia(postEntity.data);` only for posts, so the filter logic might be slightly different.

[the remark this answers is not in the document]
Comment (ice-endymion) [reply lib/app/features/feed/views/components/post/components/post_body/post_body.dart]: Moved filter to PostData

**Review by ice-alcides — APPROVED** [review]

**Review by ice-brontes — APPROVED** [review]

**Pull request merged**